samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/author/AssessmentSettingsBean.java

```diff
@@ -1177,12 +1178,15 @@ public void setRetractDateString(String retractDateString)
     if (retractDateString == null || retractDateString.trim().equals("")) {
       this.isValidRetractDate = true;
       this.retractDate = null;
+      this.isRetractAfterDue = true;
     }
     else {
 
       Date tempDate = tu.parseISO8601String(ContextUtil.lookupParam(HIDDEN_RETRACT_DATE_FIELD));
+      Date tempDueDate = tu.parseISO8601String(ContextUtil.lookupParam(HIDDEN_END_DATE_FIELD));
 
       if (tempDate != null) {
+        this.isRetractAfterDue = !(tempDueDate == null || tempDate.before(tempDueDate));
```

Comment:
I find this logic tough to understand. Is simpler to reverse the negatives?

tempDueDate != null && tempDate.after(tempDueDate)

Reply:
Sorry, I already merged it :( Would you like me to open up a new PR to change it?

Reply:
no